src/kafkaProducerManager.js

```diff
@@ -0,0 +1,36 @@
+import {KafkaProducer} from './kafkaProducer.js'
+
+export class KafkaProducerManager {
+  static kafkaSet = {}
+
+  static async getProducer(channelName, clientId, timeout) {
+    const kafkaInstance = this.findOrAddConnection(channelName, clientId, timeout)
+    if (!kafkaInstance.isConnected) await kafkaInstance.connect()
+    if (!kafkaInstance.isConnected) throw new Error('Kafka Producer failed to connect.')
+
+    return kafkaInstance.producer
+  }
+
+  static findOrAddConnection(channelName, clientId, timeout) {
+    let kafkaInstance = this.getKafkaInstance(channelName, clientId, timeout)
+    if (!kafkaInstance) {
+      kafkaInstance = new KafkaProducer(clientId, timeout)
+      this.kafkaSet[`${channelName}${clientId}${timeout}`] = kafkaInstance
```

Comment:
How about we add a urn delimiter to make easy to parse in case we need to : 
```suggestion
      this.kafkaSet[`urn:${channelName}:${clientId}:${timeout}`] = kafkaInstance
```